contents/tutorials/track-high-volume-apis.md

```diff
@@ -0,0 +1,266 @@
+
+Create a folder for our project, go into the folder, and initialize npm. When prompted, select all the default options.
+
+```bash
+mkdir high-volume
+cd high-volume
+npm init
+```
+
+Once done, install Express.
+
+```bash
+npm i express
+```
+
+In your `high-volume` folder, create a file named `index.js`, and set up a basic Express app.
+
+```node
+// index.js
+const express = require('express');
+const app = express();
+const port = 3000
+
+app.get('/', (req, res) => {
+  res.send('Hello World!');
+})
+
+app.listen(port, () => {
+  console.log(`App listening on http://localhost:${port}`)
+})
+```
+
+Lastly, create one more route as our high volume requests route. We set this route as  `/big/:id`. For now, it returns a message using the request ID.
+
+```node
+//...
+app.get('/big/:id', (req, res) => {
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+### Adding PostHog
+
+Next, we install and set up PostHog. 
+
+```bash
+npm i posthog-node
+```
+
+We set up the PostHog client with our project API key and instance address, both of which you can get from your [project settings](https://app.posthog.com/project/settings). We add a `client.capture()` call to our `/big` route. The capture call includes the route distinct ID and an event name. We also include a `firstRouteCalled` user property to use for a filter later. 
+
+```node
+const express = require('express');
+const { PostHog } = require('posthog-node');
+const app = express();
+const port = 3000
+
+const client = new PostHog(
+  '<ph_project_api_key>',
+  { host: '<ph_instance_address>'}
+)
+
+//...
+
+app.get('/big/:id', (req, res) => {
+  client.capture(
+    { 
+      event: 'big route',
+      distinctId: req.params.id,
+			properties: {
+        $set_once: { firstRouteCalled: req.url },   
+      },
+    },
+  );
+  res.send(`Welcome to the big route, ${req.params.id}`);
+})
+//...
+```
+
+When we go to the big route again, we capture an event that we can see in our PostHog instance.
+
+![Event](../images/tutorials/track-high-volume-apis/event.png)
+
+In our situation, capturing every single request to this route creates many requests. It hurts the performance of our API to make a capture request every time. Instead, we will use two different strategies for capturing them: sampling and batching.
+
+## 1. Sampling with feature flags and local evaluation
+
+We can use feature flags to sample users. This way we can easily control what amount of events we capture. For example, we can set up a flag to only capture specific or a percentage of:
+
+- Users
+- Cohorts
+- Groups
+
+To set this up, we must create a feature flag. In PostHog, go to the feature flags tab, click new feature flag, add a key (I chose "sample"), select a rollout out (I chose "100% of users where `firstRouteCalled` equals `/big/1`"), and click save. At any point, we can edit this flag to change the number of events we capture (sample). 
```

Comment:
1. Sentence is a bit long, consider breaking up into points

2.  I dont think you need to say what you did

If you'd still like to include it, for this sentence:
` (I chose "100% of users where `firstRouteCalled` equals `/big/1`")`
It's a bit unclear to me what this is or how to set it. So worth further explanation or attaching a screenshot to clarify

```suggestion
To set this up, we must create a new feature flag:

1. In PostHog, go to the feature flags tab, click new feature flag.
2. Give your key a name 
3. Select your rollout conditions – e.g. 100% of users in a specific cohort.
4. Click save. 

The main benefit of this setup is that we can edit this flag at any time to change to change the number of events we capture. 
```